test/benchee/benchmark/runner_test.exs

```diff
@@ -0,0 +1,217 @@
+defmodule Benchee.Benchmark.RunnerTest do
+  use ExUnit.Case, async: true
+  import Benchee.TestHelpers
+  alias Benchee.Suite
+  alias Benchee.Benchmark
+  alias Benchee.Configuration
+  alias Benchee.Statistics
+  alias Benchee.Test.FakeBenchmarkPrinter, as: TestPrinter
+
+  @config %Configuration{parallel: 1,
+                         time:     40_000,
+                         warmup:   20_000,
+                         inputs:   nil,
+                         print:    %{fast_warning: false, configuration: true}}
+  @system %{elixir: "1.4.0", erlang: "19.1"}
+  @default_suite %Suite{configuration: @config, system: @system}
+
+  defp test_suite(suite_override \\ %{}) do
+    DeepMerge.deep_merge(@default_suite, suite_override)
+  end
+
+  defp run_times_for(suite, job_name, input_name \\ Benchmark.no_input()) do
+    filter_fun = fn(scenario) ->
+      scenario.job_name == job_name && scenario.input_name == input_name
+    end
+    map_fun = fn(scenario) -> scenario.run_times end
+
+    suite.scenarios
+    |> Enum.filter_map(filter_fun, map_fun)
+    |> List.flatten()
+  end
+
+  describe ".run_scenarios" do
+    test "runs a benchmark suite and enriches it with measurements" do
+      retrying fn ->
+        suite = test_suite(%Suite{configuration: %{time: 60_000, warmup: 10_000}})
+        new_suite =
+          suite
+          |> Benchmark.benchmark("Name", fn -> :timer.sleep(10) end)
+          |> Benchmark.measure(TestPrinter)
+
+        assert new_suite.configuration == suite.configuration
+        run_times = run_times_for(new_suite, "Name")
+
+        # should be 6 but gotta give it a bit leeway
+        assert length(run_times) >= 5
+      end
+    end
+
+    test "runs a suite with multiple jobs and gathers results" do
+      retrying fn ->
+        suite = test_suite(%Suite{configuration: %{time: 100_000, warmup: 10_000}})
+        new_suite =
+          suite
+          |> Benchmark.benchmark("Name", fn -> :timer.sleep(19) end)
+          |> Benchmark.benchmark("Name 2", fn -> :timer.sleep(9) end)
+          |> Benchmark.measure(TestPrinter)
+
+        # should be 5 but gotta give it a bit leeway
+        assert length(run_times_for(new_suite, "Name")) >= 4
+        # should be ~11, but gotta give it some leeway
+        assert length(run_times_for(new_suite, "Name 2")) >= 8
+      end
+    end
+
+    test "can run multiple benchmarks in parallel" do
+      suite = test_suite(%Suite{configuration: %{parallel: 6, time: 60_000}})
```

Comment:
now that I see it here, 6 seems like overkill. Sticking to just 4 should work just as fine, no biggy and not related though :D